app/services/migrate_user_service.rb

```diff
@@ -0,0 +1,101 @@
+class MigrateUserService
+  def initialize(new_id:, old_id:)
+    @new_id = new_id
+    @old_id = old_id
+    @lines = []
+  end
+
+  def schema
+    {
+      visits: :user_id,
+      ahoy_events: :user_id,
+      ahoy_messages: :user_id,
+      groups: :creator_id,
+      memberships: [:user_id, :inviter_id],
```

Comment:
That's why we catch errors and keep each SQL in a seperate transaction. So in the case where there is a constraint violation, we can safely ignore it.